test/python/transpiler/test_high_level_synthesis.py

```diff
@@ -2607,5 +2607,57 @@ def test_annotated_mcx(self):
         self.assertEqual(Operator(qc), Operator(qct))
 
 
+@ddt
+class TestPauliEvolutionSynthesisPlugins(QiskitTestCase):
+    """Tests related to plugins for PauliEvolutionGate."""
+
+    def test_supported_names(self):
+        """Test that "default" and "rustiq" plugins do exist."""
+        supported_plugin_names = high_level_synthesis_plugin_names("PauliEvolution")
+        self.assertIn("default", supported_plugin_names)
+        self.assertIn("rustiq", supported_plugin_names)
+
+    @data("default", "rustiq")
+    def test_correctness(self, plugin_name):
+        """Test that all plugins return a correct Operator"""
+        op = SparsePauliOp(["XXX", "YYY", "IZZ", "XZY"], coeffs=[1, 2, 3, 4])
+        qc = QuantumCircuit(6)
+        qc.append(PauliEvolutionGate(op), [1, 2, 4])
+        hls_config = HLSConfig(PauliEvolution=[plugin_name])
+        hls_pass = HighLevelSynthesis(hls_config=hls_config)
+        qct = hls_pass(qc)
+        self.assertEqual(Operator(qc), Operator(qct))
+
+    def test_rustiq_options(self):
+        """Test important non-default Rustiq options."""
+        op = SparsePauliOp(["XXXX", "YYYY", "ZZZZ"], coeffs=[1, 2, 3])
+        qc = QuantumCircuit(6)
+        qc.append(PauliEvolutionGate(op), [1, 2, 3, 4])
+
+        # These calls to Rustiq are deterministic.
+        # On the one hand, we may need to change these tests if we switch
+        # to a newer version of Rustiq that implements different heuristics.
+        # On the other hand, these tests serve to show that the options
+        # have the desired effect of reducing the number of CX-gates.
+        with self.subTest("default_options"):
+            hls_config = HLSConfig(PauliEvolution=[("rustiq", {"upto_phase": False})])
+            hls_pass = HighLevelSynthesis(hls_config=hls_config)
+            qct = hls_pass(qc)
+            cnt_ops = qct.count_ops()
+            self.assertEqual(cnt_ops["cx"], 10)
+        with self.subTest("upto_phase"):
+            hls_config = HLSConfig(PauliEvolution=[("rustiq", {"upto_phase": True})])
+            hls_pass = HighLevelSynthesis(hls_config=hls_config)
+            qct = hls_pass(qc)
+            cnt_ops = qct.count_ops()
+            self.assertEqual(cnt_ops["cx"], 9)
```

Comment:
Could we check that the decomposition is correct for the `upto_phase` decomposition? `Operator.equiv` should be the right method 🙂

Reply:
Done.